### DISCUSSION TITLE
Discount applied to lines

### DISCUSSION THREAD

**Discussion by afewyards** [saleor/saleor#11558] [Q&A] [answered]
Currently all the coupon or order discounts are not applied to the individual lines of product orders. This leads to issues with shipping companies as they want to have the prices applied to all the individual products.

**Comment by korycins** [comment, marked as answer]
Can you provide more details? What version of Saleor do you use? What type of tax calculations do you use? Some screenshots from the dashboard with discount configuration would be good to provide. Payload that Saleor sends would be also good to provide.

As far as I see (3.9, flat rates), Saleor propagates the discount amount over a checkout/order lines
Checkout with a discount of 25$ (fixed amount, all products). Without any active Sales:
![image](https://user-images.githubusercontent.com/23282005/209347890-d94a8c20-d538-454a-9e65-6338bd28f5bb.png)

Order with manual discount 5$ attached:
![image](https://user-images.githubusercontent.com/23282005/209348108-3129fe6d-20d9-476a-99b1-ba5a9132376a.png)

https://github.com/saleor/saleor/pull/9325 will give you the option to build your own discount app and have more possibilities to customize the way how discounts work.

**Reply by afewyards** [reply]
We were using 3.7 and indeed after the upgrade this has resolved. But I would assume a change like this would be more noticeable documented on the changelog

With the flat rates we noticed that all the discount codes values are now in excl tax. They should imho always be with included tax.
